clang/lib/Sema/SemaConcept.cpp

```diff
@@ -971,9 +971,12 @@ static const Expr *SubstituteConstraintExpressionWithoutSatisfaction(
   // this may happen while we're comparing two templates' constraint
   // equivalence.
   LocalInstantiationScope ScopeForParameters(S);
-  if (auto *FD = DeclInfo.getDecl()->getAsFunction())
-    for (auto *PVD : FD->parameters())
-      ScopeForParameters.InstantiatedLocal(PVD, PVD);
+  if (const NamedDecl *D = DeclInfo.getDecl()) {
+    const FunctionDecl *FD = D->getAsFunction();
```

Comment:
When is D null?

Reply:
(I roughly explained in the comment.)
This function `SubstituteConstraintExpressionWithoutSatisfaction` would be called in the context where a template parameter list is getting parsed, when the template declaration that the template parameter list would attach to is not yet available.